src/config.rs

```diff
@@ -274,6 +274,15 @@ fn parse_proxy_config() -> Result<ProxyConfig, Error> {
 
 pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
     let ipv6_enabled = parse::<bool>(IPV6_ENABLED)?.unwrap_or(true);
+    let ipv6_localhost_enabled = if ipv6_enabled {
```

Comment:
I actually think `ipv6_localhost_enabled` _shouldn't_ be proxywide static config, since it's not, really - it's context-sensitive.

I actually prefer doing the check in `socket_factory` at bind-time or close to it with helpers, rather than trying to have a canonical config value for it.

Can we just do that, and remove `ipv6_localhost_enabled` from the config object?

Reply:
`ipv6_localhost_enabled` never actually ends up into `Config` fwiw. Its just a local var. At the time I think this made a lot of sense; I did not want to check it on each access since the check is not free (its pretty quick, but still).

But with the DNS change its a bit murkier. In some cases, we have a proxy-wide config (admin), but in others we have per-proxy. In the proxy-wide cases, we don't even have a socket_factory at all

Reply:
> ipv6_localhost_enabled never actually ends up into Config fwiw. Its just a local var. At the time I think this made a lot of sense; I did not want to check it on each access since the check is not free (its pretty quick, but still).

Ah right - I was thinking it might make more sense to actually put it in `admin.rs` as a helper, since that is the only place we do a proxywide bind.

> But with the DNS change its a bit murkier. In some cases, we have a proxy-wide config (admin), but in others we have per-proxy. In the proxy-wide cases, we don't even have a socket_factory at all

Yes that's fair, I was thinking that since we don't use it as a nonrecomputed var anywhere but `admin` it maybe should live in `admin.rs` or similar as a local opinion for that, and the fact that we hard-check it again in the DNS server anyway makes me think the recomputed version might be better kept in `socket_factory` - where we could use it for pods too, right?

Reply:
Well technically we have 4 things using Address today, just only 2 of them generally are bound to localhost. The obvious way to defer the "use ipv6" is to make into_iter() take a "supports v6" flag, but then you need to always pass that on all calls.

That is still only 3+N calls, though. I am fine either way, WDYT?

Reply:
not a big deal, yeah.